src/reducers/MapReducer.js

```diff
@@ -61,7 +129,14 @@ export default (state = INITIAL_STATE, action) => {
             const {region, sites} = payload;
             const existingMapLoadedState = state.mapLoaded;
 
-            return payload ? {...INITIAL_STATE, lastKnownRegion: region, sites, mapLoaded: existingMapLoadedState} : INITIAL_STATE;
+            return payload ? {
+                ...INITIAL_STATE,
+                lastKnownRegion: region,
+                sites: sites,
+                filterCriteriaKeys: state.filterCriteriaKeys,
+                displaySites: filterSites({sites}, state.filterCriteriaKeys),
+                mapLoaded: existingMapLoadedState
+            } : INITIAL_STATE;
```

Comment:
Had to change our initial site loading logic now that we have filtering. We still keep track of all possible sites in `sites`, but what we send to the different screens will be a filtered version called `displaySites`. When we load new sites we need to also subject them to any current filtering, and set it all to state.